pkg/network/infraconfigurators/masquerade_test.go

```diff
@@ -0,0 +1,148 @@
+package infraconfigurators
+
+import (
+	"fmt"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vishvananda/netlink"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	"kubevirt.io/kubevirt/pkg/network/cache/fake"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+var _ = Describe("Masquerade infrastructure configurator", func() {
+	var (
+		cacheFactory cache.InterfaceCacheFactory
+		ctrl         *gomock.Controller
+		handler      *netdriver.MockNetworkHandler
+	)
+
+	BeforeEach(func() {
+		cacheFactory = fake.NewFakeInMemoryNetworkCacheFactory()
+		ctrl = gomock.NewController(GinkgoT())
+		handler = netdriver.NewMockNetworkHandler(ctrl)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	const (
+		bridgeIfaceName = "br0"
+	)
+
+	newVMIMasqueradeInterface := func(namespace string, name string) *v1.VirtualMachineInstance {
+		vmi := v1.NewMinimalVMIWithNS(namespace, name)
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMasqueradeNetworkInterface()}
+		v1.SetObjectDefaults_VirtualMachineInstance(vmi)
+		return vmi
+	}
+
+	Context("discover link information", func() {
```

Comment:
Not sure we need this nesting level since there is only one context for this Spec

Reply:
I want to individually describe each of the functions of the API.